core/src/main/java/org/apache/carbondata/core/scan/result/impl/FilterQueryScannedResult.java

```diff
@@ -81,7 +103,24 @@ public FilterQueryScannedResult(BlockExecutionInfo tableBlockExecutionInfos,
   }
 
   @Override public List<byte[][]> getNoDictionaryKeyArrayBatch(int batchSize) {
-    throw new UnsupportedOperationException("Operation not supported");
+    List<byte[][]> noDictionaryKeyArrayList = new ArrayList<>(validRowIds.size());
```

Comment:
This method also duplicated, please move to `BlockletScannedResult`

Reply:
Done.